test/cancel_test.go

```diff
@@ -135,30 +128,25 @@ func TestTaskRunPipelineRunCancel(t *testing.T) {
 	}
 
 	t.Logf("Waiting for TaskRuns in PipelineRun %s in namespace %s to be cancelled", "pear", namespace)
-	errChan2 := make(chan error, len(taskrunList.Items))
-	defer close(errChan2)
 	for _, taskrunItem := range taskrunList.Items {
-		go func(errChan2 chan error) {
-			err := WaitForTaskRunState(c, taskrunItem.Name, func(tr *v1alpha1.TaskRun) (bool, error) {
+		go func(name string) {
```

Comment:
Same here

```suggestion
		wg.Add()
		go func(name string) {
			defer wg.Done()
```

with a `wg.Wait()` outside the loop